src/AuxDidAuth.ts

```diff
@@ -37,24 +38,31 @@ import {
   DidAuthKeyCurve,
   DidAuthResponseOptsNoSignature,
 } from "./interfaces/DIDAuth.types";
-import { getPublicJWKFromPrivateHex } from "./util/JWK";
+import {
+  getPublicJWKFromPrivateHex,
+  getPublicJWKFromPublicHex,
+} from "./util/JWK";
 import { DEFAULT_PROOF_TYPE, PROOF_TYPE_EDDSA } from "./config";
 
 const isInternalSignature = (
-  object: InternalSignature | ExternalSignature
+  object: InternalSignature | ExternalSignature | NoSignature
 ): object is InternalSignature => "hexPrivateKey" in object && "did" in object;
 
 const isExternalSignature = (
-  object: InternalSignature | ExternalSignature
+  object: InternalSignature | ExternalSignature | NoSignature
 ): object is ExternalSignature => "signatureUri" in object && "did" in object;
 
+const isNoSignature = (
+  object: InternalSignature | ExternalSignature | NoSignature
+): object is InternalSignature => "hexPublicKey" in object && "did" in object;
```

Comment:
Aquí no cal posar `is NoSignature? => `

Reply:
Si tens raó, encara que retornava true igual perquè tenia hexpublicKey. El corregeixo